Move components in wp.blocks.InspectorControls.* to wp.components.*.

jorgefilipecosta opened the pull request:
The components were not inspector controls specific, they were just generic input elements. This move will allow other parts of the Gutenberg besides blocks, e.g: document sidebar to take advantage of this reusable components.
This PR comes after some discussions in https://github.com/WordPress/gutenberg/pull/4175#pullrequestreview-85666908.
This move will also allow the move of TermTreeSelect, as TreeSelect component. Making possible to use it in the HierarchicalTermSelector removing existing duplicate code.
The inputs in Document sidebar could also be updated to use the components e.g: toggles (Stick to the Front Page)  and selects (Author, Template)  could make use ToggleControl and SelectControl, instead of repeating the same logic.

We expose a select control so Fixes: https://github.com/WordPress/gutenberg/issues/3931;

Deprecation warnings are shown when importing the components from wp.blocks.InspectorControls.*, but for now, the import continues to work as before.

## How Has This Been Tested?
This refactor affects the whole application so besides executing automatic tests, do general smoke testing around the app and verify things are working as before. Give special relevance to inspector controls.

jorgefilipecosta committed:
Move components in wp.blocks.InspectorControls.* to wp.components.*.

The components were not inspector controls specific, they were just generic input elements. This move will allow other parts of the Gutenberg besides blocks, e.g: document sidebar to take advantage of this reusable components.

youknowriad commented on blocks/inspector-controls/index.js:
```diff
@@ -23,11 +27,33 @@ export default function InspectorControls( { children } ) {
+	class WrappedComponent extends Component {
+		componentDidMount() {
+			// eslint-disable-next-line no-console
+			console.warn( `wp.blocks.InspectorControls.${ componentName } is deprecated use wp.components.${ componentName }` );
```
Maybe `wp.blocks.InspectorControls.${ componentName } is deprecated, use wp.components.${ componentName }.`

Added punctuation.

youknowriad commented on blocks/inspector-controls/index.js:
```diff
@@ -23,11 +27,33 @@ export default function InspectorControls( { children } ) {
+};
+
+forEach(
+	{ BaseControl,
```
Should we move BaseControl to a new line aswell?

youknowriad reviewed (commented):
It might be a good idea to  add a small README per component (like other generic components :))

jorgefilipecosta commented:
Thank you for your review @youknowriad, this PR was rebased and the feedback was addressed.

youknowriad commented:
While I agree with this change, I added some reviewers to have other opinions here.

aduth commented on components/input-controls/range-control/index.js:
```diff
@@ -1,13 +1,15 @@
  * Internal dependencies
  */
 import BaseControl from './../base-control';
+import Button from '../../button';
```
It's a shame this doesn't work, but it's frustrated me in the past in my attempts to do so:

```js
import { Dashicon, withInstanceId, Button } from '../../';
```

jorgefilipecosta replied:
Yes, this syntax would be better :(

aduth reviewed (commented):
Not sure we need the nesting of `input-controls` within components.

I don't recall the specific reason why it was implemented the way it was originally; if we thought that the inputs were specific to usage within blocks/editor? As noted, they seem generic enough, so I don't have much of a problem with this.

jorgefilipecosta commented:
Maybe at the time, these controls were created the idea was for them to be InspectorControls specific maybe even render them automatically inside inspector controls, but that ended up not being the case and the components ended up being generic.

I rebased this PR adding some required changes meanwhile. All the feedback was addressed namely, removed the input-controls nesting and added a readme.md file for each component documenting them.

youknowriad reviewed (approved):
I like the READMEs 👍

phpbits commented:
@jorgefilipecosta seems like the Custom CSS Class is now missing. I've run build package and install the zip file. I'm not seeing the field now. Thanks!

jorgefilipecosta commented:
Thank you for reporting this problem @phpbits. I'm also able to replicate it, and I'm investigating it's cause.

phpbits commented:
@jorgefilipecosta you're very welcome. Let me know how it goes. I'm also using Inspector Control on my plugin integration and also not showing. Hope you can share the fixes. Thanks!

jorgefilipecosta commented:
The root cause was that focus prop was been used to check condition while it was removed.
A fix was submitted in PR https://github.com/WordPress/gutenberg/pull/5004.
@phpbits I hope this fixes the issue you are having applying a similar change to your plugin.

phpbits commented:
@jorgefilipecosta I'll wait for the pull request to be merged and will try it on my end. I hope this will fix my issues too. Thanks!

phpbits commented:
@jorgefilipecosta Yes fields are now showing but it's not saving. Have you tried adding values to custom CSS Class field and save the changes? Thanks!
